pkg/kv/kvserver/protectedts/protectedts.go

```diff
@@ -139,6 +141,30 @@ type Verifier interface {
 	Verify(context.Context, uuid.UUID) error
 }
 
+// Metrics encapsulates the metrics exported by the reconciler.
+type Metrics struct {
+	ReconcilationRuns    *metric.Counter
+	RecordsProcessed     *metric.Counter
+	RecordsRemoved       *metric.Counter
+	ReconciliationErrors *metric.Counter
+}
+
+var _ metric.Struct = (*Metrics)(nil)
+
+// MetricStruct makes Metrics a metric.Struct.
+func (m *Metrics) MetricStruct() {}
+
+// Reconciler provides a mechanism to reconcile protected timestamp records with
+// external state.
+type Reconciler interface {
+	// StartReconciler will start the reconciliation where each record's status is
+	// determined using the record's meta type and meta in conjunction with the
+	// configured StatusFunc.
+	StartReconciler(ctx context.Context, stopper *stop.Stopper) error
```

Comment:
Note that it is safe to call this more than once because the work is idempotent?